### PR TITLE
fix(condo): DOMA-5473 updated Select component

### PR DESCRIPTION
**Pull request by Alllex202**
## Changes

 - Updated styles
 - Added empty dropdown content
 - Added new props
 - Added multiple mode
  
Снимок экрана 2023-03-03 в 13 31 40

 - Fixed a situation where the selected long value went beyond the width of the component
 
Снимок экрана 2023-03-03 в 00 02 02

---

## Issues
 - When `displayMode` is set to `fit-content` and selected multiple mode then component to collapse. This is due to the implementation of the component in `antd`.

 #### Multiple mode
Снимок экрана 2023-03-06 в 19 01 00
 #### No multiple mode
Снимок экрана 2023-03-06 в 19 00 51

### CODE REVIEW

File: packages/ui/src/components/Select/removeIcon.tsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import React from 'react'
```
Comment (SavelevMatthew) [inline packages/ui/src/components/Select/removeIcon.tsx:1]: Should it come from `@open-condo/icons`?

File: packages/ui/src/components/Select/style.less [hunk 2]
```diff
@@ -60,20 +59,64 @@
...
-      align-items: center;
-      justify-content: flex-start;
+      display: inline-block;
       width: 100%;
       padding-right: 0;
+      overflow: hidden;
       color: @condo-global-color-black;
       font-size: @condo-global-font-size-text-m;
       font-family: @condo-global-font-family-default;
+      line-height: 46px;
```
Comment (SavelevMatthew) [inline packages/ui/src/components/Select/style.less:76]: why `size % 4 != 0`?

File: packages/ui/src/components/Select/style.less [hunk 3]
```diff
@@ -60,20 +59,64 @@
...
+      padding: 2px 0;
+    }
+
+    .condo-select-selection-item {
+      display: flex;
+      align-items: center;
+      justify-content: flex-start;
+      width: inherit;
+      height: 32px;
+      margin: 5px 10px 5px 0;
```
Comment (SavelevMatthew) [inline packages/ui/src/components/Select/style.less:95]: 5?

File: packages/ui/src/components/Select/select.tsx [hunk 4]
```diff
@@ -64,17 +77,33 @@ const Select: ISelect = (props) => {
...
-                    <Option key={optionIndex} value={value} disabled={disabled}>
-                        <Typography.Text size='medium' disabled={disabled} type={textType}>
-                            {label}
-                        </Typography.Text>
+                ? options.map(({ label, value, textType, disabled, title }, optionIndex) => (
+                    <Option
+                        key={optionIndex}
+                        value={value}
+                        disabled={disabled}
+                        title={(title || typeof label !== 'string') ? title : label}
```
Comment (akarjakin) [inline packages/ui/src/components/Select/select.tsx:104]:
```
title={title || !isString(label) ? title : label}
```

**Review by akarjakin — APPROVED** [review]

File: packages/ui/src/components/Select/select.tsx (hunk 4 above)
Comment (SavelevMatthew) [reply packages/ui/src/components/Select/select.tsx:104]: `... || ... ? ... : ...` without grouping condition seems a bit more dirty to me

File: packages/ui/src/components/Select/style.less [hunk 5]
```diff
@@ -4,76 +4,167 @@
...
+      width: calc(100% - @select-padding-right - @select-padding-left);
       padding-right: 0;
+      overflow: hidden;
       color: @condo-global-color-black;
+      font-weight: @condo-global-typography-text-m-font-weight;
       font-size: @condo-global-font-size-text-m;
-      font-family: @condo-global-font-family-default;
+      font-family: @condo-global-typography-text-m-font-family;
+      line-height: @condo-global-typography-text-m-line-height;
+      letter-spacing: @condo-global-typography-text-m-letter-spacing;
```
Comment (SavelevMatthew) [inline packages/ui/src/components/Select/style.less:82]: How about typography mixin here? (`.condo-typography-text-medium()`)

**Review by SavelevMatthew — COMMENTED** [review]
LGTM, but multiple is not stackable with type?

File: packages/ui/src/components/Select/select.tsx [hunk 6]
```diff
@@ -21,26 +21,39 @@ export type OptionType = React.PropsWithChildren<{
...
-}
+const Select = <ValueType extends SelectValueTypeBase>(props: SelectProps<ValueType>): React.ReactElement => {
+    const { options, children, displayMode = 'fill-parent', type, onChange, id, isMultiple, notFoundContentLabel,  ...rest } = props
 
-const Select: ISelect = (props) => {
-    const { options, children, displayMode = 'fill-parent', type, onChange, id, ...rest } = props
     const className = classNames({
         [`${SELECT_CLASS_PREFIX}-${displayMode}`]: displayMode,
-        [`${SELECT_CLASS_PREFIX}-${type}`]: type,
+        [`${SELECT_CLASS_PREFIX}-${type}`]: !isMultiple && type,
```
[the remark this answers is not in the document]
Comment (Alllex202) [reply packages/ui/src/components/Select/select.tsx:56]: This happens automatically inside `ant`

File: packages/ui/src/components/Select/style.less [hunk 7]
```diff
@@ -133,12 +280,40 @@
   border-radius: @condo-global-border-radius-medium;
   box-shadow: @condo-global-box-shadow-default;
 
+  .condo-select-item-empty {
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    padding: 20px;
+
+    * {
```
[the remark this answers is not in the document]
Comment (Alllex202) [reply packages/ui/src/components/Select/style.less:289]: fixed it

**Comment by Alllex202** [comment]
> LGTM, but multiple is not stackable with type?

We do not have such a need. If it is ever needed, it can be added.

`type` is specific to statuses only, nowhere else

**Pull request merged**